codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerBuilderSymbol.kt

```diff
@@ -17,13 +17,23 @@ import software.amazon.smithy.rust.codegen.core.smithy.rustType
 import software.amazon.smithy.rust.codegen.core.util.toSnakeCase
 import software.amazon.smithy.rust.codegen.server.smithy.ServerCodegenContext
 
+val alreadyDefined : HashMap<String, RustModule.LeafModule> = HashMap()
```

Comment:
Perhaps this should be `ConcurrentHashMap` like the one in `RustCrateInlineModuleComposingWriter`.

Reply:
Actually, why does this need to be a map? Can't it be a set of `RustModule.LeafModule`s?

Reply:
Coming back to this again, why do we need to memoize the modules at all? Can't `serverBuilderModule` simply always calculate and return `RustModule.new(builderNamespace, visibility, parent = structureSymbol.module(), inline = true)`?

Reply:
Good catch. Actually, this `HashMap` thing was never to be committed. At that time, I thought the reason it keeps creating new inline modules was that the code keeps on calling `RustModule.new` instead of returning the same inline module object.

Have fixed it and now it returns `RustModule.new(...)`